src/script/op_codes.rs

```diff
@@ -36,92 +38,301 @@ pub fn to_boxed_hash(slice: &[u8]) -> Box<HashValue> {
+    msg.copy_from_slice(slice);
+    msg
+}
 
-// Opcode constants: Script termination
-pub const OP_RETURN: u8 = 0x60;
+/// Convert a slice into a Boxed Message
+pub fn to_boxed_message(slice: &[u8]) -> Box<Message> {
+    Box::new(to_message(slice))
+}
 
-// Opcode constants: Stack manipulation
+/// Convert a slice of little endian bytes into a u64.
+///
+/// # Panics
+///
+/// The function does not check slice for length at all.  You need to check this / guarantee it yourself.
+pub fn to_u64(slice: &[u8]) -> u64 {
+    let mut num = [0u8; 8];
+    num.copy_from_slice(slice);
+    u64::from_le_bytes(num)
+}
+
+/// Convert a slice of little endian bytes into an i64.
+///
+/// # Panics
+///
+/// The function does not check slice for length at all.  You need to check this / guarantee it yourself.
+pub fn to_i64(slice: &[u8]) -> i64 {
+    let mut num = [0u8; 8];
+    num.copy_from_slice(slice);
+    i64::from_le_bytes(num)
+}
+
+// Opcode constants: Block Height Checks
+pub const OP_CHECK_HEIGHT_VERIFY: u8 = 0x66;
+pub const OP_CHECK_HEIGHT: u8 = 0x67;
+pub const OP_COMPARE_HEIGHT_VERIFY: u8 = 0x68;
+pub const OP_COMPARE_HEIGHT: u8 = 0x69;
+
+// Opcode constants: Stack Manipulation
 pub const OP_DROP: u8 = 0x70;
 pub const OP_DUP: u8 = 0x71;
 pub const OP_REV_ROT: u8 = 0x72;
 pub const OP_PUSH_HASH: u8 = 0x7a;
 pub const OP_PUSH_ZERO: u8 = 0x7b;
+pub const OP_NOP: u8 = 0x73;
+pub const OP_PUSH_ONE: u8 = 0x7c;
+pub const OP_PUSH_INT: u8 = 0x7d;
+pub const OP_PUSH_PUBKEY: u8 = 0x7e;
 
-// Opcode constants: Comparisons
+// Opcode constants: Math Operations
 pub const OP_EQUAL: u8 = 0x80;
 pub const OP_EQUAL_VERIFY: u8 = 0x81;
-
-// Opcode constants: Arithmetic
 pub const OP_ADD: u8 = 0x93;
 pub const OP_SUB: u8 = 0x94;
+pub const OP_GE_ZERO: u8 = 0x82;
+pub const OP_GT_ZERO: u8 = 0x83;
+pub const OP_LE_ZERO: u8 = 0x84;
+pub const OP_LT_ZERO: u8 = 0x85;
 
-// Opcode constants: Cryptography
+// Opcode constants: Boolean Logic
+pub const OP_OR_VERIFY: u8 = 0x64;
+pub const OP_OR: u8 = 0x65;
+
+// Opcode constants: Cryptographic Operations
 pub const OP_CHECK_SIG: u8 = 0xac;
 pub const OP_CHECK_SIG_VERIFY: u8 = 0xad;
 pub const OP_HASH_BLAKE256: u8 = 0xb0;
+pub const OP_HASH_SHA256: u8 = 0xb1;
+pub const OP_HASH_SHA3: u8 = 0xb2;
+
+// Opcode constants: Miscellaneous
+pub const OP_RETURN: u8 = 0x60;
+pub const OP_IF_THEN: u8 = 0x61;
+pub const OP_ELSE: u8 = 0x62;
+pub const OP_END_IF: u8 = 0x63;
 
 #[derive(Debug, Clone, PartialEq, Eq)]
 pub enum Opcode {
-    /// Push the current chain height onto the stack
-    PushHeight,
-    /// Push the associated 32-byte value onto the stack
-    PushHash(Box<HashValue>),
-    /// Push a zero onto the stack
+    // Block Height Checks
+    /// Compare the current block height to height. Fails with VERIFY_FAILED if the block height < height.
+    CheckHeightVerify(u64),
+    /// Pushes the value of (the current height - height) to the stack. In other words, the top of the stack will hold
```

Comment:
I also don't have a strong opinion on this one, but if its typically called `depth`, we would probably benefit to changing this
But on the comments, I think we should change the one height to tip to make it less ambiguous 
```suggestion
    /// Pushes the value of (the current tip height - height) to the stack. In other words, the top of the stack will hold
```
.

Reply:
Good point will do